### PR TITLE
Fixup some REST-related logs

### PR DESCRIPTION
**Pull request by tristan957**
Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

File: lib/kvdb/kvdb_rest.c [hunk 1]
```diff
@@ -871,38 +871,54 @@ kvdb_rest_add_endpoints(struct ikvdb *const kvdb)
     alias = ikvdb_alias(kvdb);
 
     status = rest_server_add_endpoint(0, handlers[0], kvdb, "/kvdbs/%s/compact", alias);
-    if (ev(status) && !err)
+    if (ev(status) && !err) {
```
Comment (tristan957) [inline lib/kvdb/kvdb_rest.c:874]: If this fails, we bring down all of HSE. Presumably, this is not what we want. Do others agree?

Comment (alexttx) [reply lib/kvdb/kvdb_rest.c:874]: It's called during `kvdb_open`, and most of the failure cases seem to be developer error (sprintf overflow, duplicated enpoints, etc).  I would say failing the `kvdb_open` call is the correct action.

Comment (tristan957) [reply lib/kvdb/kvdb_rest.c:874]: If you look in hse_init(), you can see I am ignoring errors. Should I stop doing that then?

**Comment by tristan957** [comment]
This is partially related to NFSE-5377.

File: lib/kvdb/kvdb_rest.c (hunk 1 above)
Comment (alexttx) [reply lib/kvdb/kvdb_rest.c:874]: As currently coded, these failures will not cause opens to fail.  You'd have to change `ikvdb_rest_register` to pass the error to `ikvdb_open`.

Comment (tristan957) [reply lib/kvdb/kvdb_rest.c:874]: Nice catch...I will submit a separate PR to fail things then.

**Review by alexttx — APPROVED** [review]

**Pull request merged**